hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCreateTransitionLogic.java

```diff
@@ -150,8 +150,11 @@ public void doStateTransitionOperation(
             final Id relayerId,
             final long maxGasAllowance,
             final BigInteger userOfferedGasPrice) {
+        worldState.clearProvisionalContractCreations();
```

Comment:
I know nothing about how this works, so as a result I have questions:

1) why isn't it that instead of a really specific call "clearProvisionalContractCreations" there isn't a more general "resetForNewTx" or maybe "resetAfterTxComplete" that would include this, and possibly other operations too?  And then, maybe there's a better place to put it than at the two different "doStateTransitionOperations".  It just seems that this is kind of low level and perhaps a higher level "concept" is what's needed and then there'd be a single obvious place to put it.

2) Isn't this "clear" needed because the state is being updated at various points?  And the "clear" aka "reset" needs to be done between those updates? And aren't those state updates always handled by some kind of "updater"?  So maybe the right place to do this kind of reset is when the _first_ updater (of a stack) is created off of the world state?  (Assuming, for lack of knowledge on my part, that there's only ever _one_ updater stack working at a time

Let me know how wrong I am....

Reply:
Here's my understanding:
1. That's a good point, I think the idea is to fix the specific issue that we know of now and have a more general implementation at a later stage (probably not in this milestone) that should prevent against such issues.
2. The updates were part of the commit logic of the world state and previously we had them cleared every time we read the needed updates, but with the changes introduced around hollow account creation we had the commit logic write the updates (which it should not for non-contract accounts) and then we did not read those updates because we don't need them but it meant that we will not clear them from the state. So the various points the updates are done are the state commits and the clear is done between them at the start of each transaction logic.

cc @Nana-EC @tinker-michaelj

Reply:
ok on both.  your answer to my question 1 indicates you think there might be some technical debt accrued here - let's not forget it!